deployment/ccip/changeset/changeset_operations.go

```diff
@@ -0,0 +1,227 @@
+package changeset
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/chainlink/deployment"
+	deployment_ethereum "github.com/smartcontractkit/chainlink/deployment/ethereum/extension"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/fee_quoter"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/nonce_manager"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/offramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/onramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/rmn_home"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/rmn_remote"
+)
+
+type DeployChainInput struct {
+	// Targeted chain
+	ChainID uint64
+
+	// Existing contract addresses
+	Weth9ContractAddress      common.Address
+	RMNHomeAddress            common.Address
+	LINKAddress               common.Address
+	TokenAdminRegistryAddress common.Address
+	RMNProxyAddress           common.Address
+	MockRMNAddress            common.Address
+	RMNAddress                common.Address
+	TimelockAddress           common.Address
+
+	// Params needed for new deployments and configs
+	FeeQuoterParams FeeQuoterParams
+	OffRampParams   OffRampParams
+}
+
+/*
+Replication of deployChainContractsEVM changeset in cs_deploy_chain.go
+
+Process:
+
+- Verify some contracts exist in the state: LINK...
+- rmn_remote.DeployRMNRemote()
+- activeDigest, err := rmnHome.GetActiveDigest(&bind.CallOpts{})
+- rmnRemoteContract.SetConfig()
+- router.DeployRouter()
+- nonce_manager.DeployNonceManager()
+- fee_quoter.DeployFeeQuoter()
+- onramp.DeployOnRamp()
+- offramp.DeployOffRamp()
+- feeQuoterContract.ApplyAuthorizedCallerUpdates()
+*/
+var DeployChain = func(env deployment.Environment, input DeployChainInput) (deployment.ChangesetOutput, error) {
```

Comment:
This changeset replicates the logic in the original [changeset](https://github.com/smartcontractkit/chainlink/blob/develop/deployment/ccip/changeset/cs_deploy_chain.go#L283C1-L539C2)

Following the Operation pattern, we get default logging, auditing, reporting, and idempotency without developers needing to explicitly customize their logic for it